### PR TITLE
update?

### PR DESCRIPTION
**Pull request by SuitedTom**

### CODE REVIEW

**Comment by SuitedTom** [comment]
should be good though you should review

File: commands/banner.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    required_bot_permissions = "SEND_MESSAGES",
+)]
+pub async fn submit_image(
+    ctx: Context<'_>,
+    #[description = "Image to submit"] image: AttachmentType,
+) -> Result<(), Error> {
+    let guild_id = ctx.guild_id().ok_or(CommandErr::GuildOnly)?;
+
+    // Define the private channel ID where submissions will be sent
+    let private_channel_id = ChannelId(123456789012345678); // Replace with your private channel ID
```
Comment (NotNorom) [inline commands/banner.rs:17]: This bot is designed to run in multiple guilds so a hard coded value at this point would not work :<

File: commands/banner.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    required_permissions = "MANAGE_GUILD",
+)]
+pub async fn approve_image(
+    ctx: Context<'_>,
+    #[description = "Image URL to approve"] image_url: String,
+) -> Result<(), Error> {
+    let guild_id = ctx.guild_id().ok_or(CommandErr::GuildOnly)?;
+
+    // Add the approved image to the rotation (implement your own storage for approved images)
+    // For simplicity, we'll just log it here
```
Comment (NotNorom) [inline commands/banner.rs:57]: I currently do not plan on storing the images themselves anywhere and instead just using the images hosted on discord.

**Review by NotNorom — COMMENTED** [review]
Hey so this is a nice idea but I cannot currently merge this because of the things I mentioned in the code.

I do like the idea of having users submit an image and them being approved! With the changes introduced since the v0.5.1 tag this will be possible by having a discord channel full of approved images and a discord channel full of image requests :)

File: src/bin/bot.rs [hunk 3]
```diff
@@ -53,6 +54,14 @@ async fn main() -> Result<(), Error> {
...
     .await?;
 
+    // Spawn a task to handle SIGINT
+    let shard_manager = client.shard_manager.clone();
+    tokio::spawn(async move {
+        signal::ctrl_c().await.expect("Failed to listen for ctrl_c");
+        shard_manager.lock().await.shutdown_all().await;
+        info!("Received SIGINT, shutting down.");
+    });
+
```
Comment (NotNorom) [inline src/bin/bot.rs:68]: I do like and and I want to have it as you can see from #14

File: commands/banner.rs (hunk 1 above)
Comment (SuitedTom) [reply commands/banner.rs:17]: Ah yes forgot about that, I'll work on changing it

File: commands/banner.rs (hunk 2 above)
Comment (SuitedTom) [reply commands/banner.rs:57]: Ah

**Comment by NotNorom** [comment]
If you want, we can talk about what you need from the bot in my support discord: https://discord.gg/MMJFtCtYPP

Maybe we can figure stuff out together :)

**Comment by NotNorom** [comment]
Closed because of inactivity. Can be reopened when needed.

**Pull request closed without merging**